### PR TITLE
reap the exit statuses from 'onevent' child processes using futures

### PR DESCRIPTION
**Pull request by wfraser**
A better fix for #209 than #221 or #254; this uses futures rather than blocking the main thread or spawning a thread for each process.

### CODE REVIEW

**Comment by wfraser** [comment]
Note that this does not attempt to kill children after some timeout or anything like that. If you spawn programs from librespot that run forever, you deserve what happens :)

File: src/main.rs [hunk 1]
```diff
@@ -466,7 +469,16 @@ impl Future for Main {
             if let Some(ref mut player_event_channel) = self.player_event_channel {
                 if let Async::Ready(Some(event)) = player_event_channel.poll().unwrap() {
                     if let Some(ref program) = self.player_event_program {
-                        run_program_on_events(event, program);
+                        let child = run_program_on_events(event, program)
+                            .expect("program failed to start");
```
Comment (wfraser) [inline src/main.rs]: note: this is bad behavior and should probably be handled some other way than panicking, but it's what the existing code does...

**Comment by erkia** [comment]
Doesn't work for me - no zombies are reaped.

**Comment by wfraser** [comment]
Ah you're right, my future is flawed; it doesn't re-register itself with the event loop. But I notice there is a tokio crate that implements this properly in much the same way. New commit coming.

**Comment by michalfita** [comment]
Shouldn't the title of this PR contain "Fix #209:"?

**Comment by erkia** [comment]
This fix now works for me and I hope it gets merged soon.

**Comment by sashahilton00** [comment]
Tested, works here as well. Open an issue if subsequent problems discovered.

**Pull request merged**